src/main/daml/Daml/Finance/Lifecycle/Event/DateClockUpdate.daml

```diff
@@ -1,27 +1,27 @@
 -- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
 
-module Daml.Finance.Lifecycle.Event.DateClock where
+module Daml.Finance.Lifecycle.Event.DateClockUpdate where
 
+import Daml.Finance.Interface.Types.Common (Id, Parties)
 import Daml.Finance.Interface.Lifecycle.Event qualified as Event (I, View(..))
 import Daml.Finance.Interface.Lifecycle.Event.Time qualified as Time (HasImplementation, I, View(..))
-import Daml.Finance.Interface.Types.Common (Id, Parties)
-import Daml.Finance.Interface.Types.Date.Classes (HasUTCTimeConversion(..))
-import Daml.Finance.Lifecycle.Types (Unit)
 
--- | Type synonym for `DateClock`.
-type T = Event
+-- | Type synonym for `DateClockUpdateEvent`.
+type T = DateClockUpdateEvent
 
-instance Time.HasImplementation Event
+instance Time.HasImplementation DateClockUpdateEvent
 
--- | Event signalling the update of a (local date) clock. It can trigger the execution of lifecycle
--- rules for some instruments.
-template Event
+-- | Event signalling the update of a `DateClock`. It can trigger the execution of lifecycle rules
+-- for some instruments.
+template DateClockUpdateEvent
   with
     providers : Parties
       -- ^ Providers of the event.
-    date : Unit
+    date : Date
```

Comment:
The core change is here.

The `Unit` type arbitrarily converts a `Date` to UTC noon. This should not trickle to a generic event.